lib/ccflay.rb

```diff
@@ -71,6 +71,14 @@ def initialize(option = nil)
   Sexp::NODE_NAMES[name] = Sexp::NODE_NAMES.size
 end
 
+# Overwrite method from Flay to suppress ignorable errors and for thread safety
+# (please see bug #206)
+node_names_mutex = Mutex.new
```

Comment:
Is this sufficient? I could be wrong, but I thought that we would need to initialize the mutex before spawning threads so the same mutex is used in each thread

Reply:
Great question. Definitely a little odd-looking, but yes, it's sufficient.

This file is unconventional in a few ways, one of them being some code is evaluated when the file _itself_ loaded (as opposed to a `class` or `module` _in_ the file). Here's a [link](https://github.com/codeclimate/codeclimate-duplication/blob/f9dff44d0064168107b4ee1f02214f24dc3daa4c/lib/ccflay.rb#L74-L76) to see it in greater context.

Reply:
Re: Max's point: I think this gets initialized once & only once when the class gets declared, so there's only one instance ever, so it's correct as far as the threading goes.

Stylistically, I find the pattern of local-vars-in-class-decls super weird & a bit of a smell. Could this be a constant instead? Would there be a downside to that stylistic change?

Reply:
ah yes the context helps a lot. I was just assuming this was running more than once. But it will be the same mutex each time that lambda is called. Neat, thanks for explaining 👍

Reply:
> Could this be a constant instead? Would there be a downside to that stylistic change?

Yup! I think I was just following the lead of the `new_nodes` assignment.

I think the downside is that the mutex, by virtue of being a constant, would be globally scoped, when it's really only needed here.